src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,293 @@
+                return;
+            }
+
+            if (participantRoles.contains(Role.STUDY_COORDINATOR) && orgMembership == null) {
+                LOG.info("User is a Study Coordinator and does not have an org membership.");
+                return;
+            }
+
+            if (studyId != null && (participantRoles.contains(Role.STUDY_COORDINATOR) ||
+                    (participantRoles.contains(Role.RESEARCHER) && orgMembership != null))) {
+                boolean sponsored = false;
+                List<Study> studies = bridgeHelper.getSponsoredStudiesForApp(appId, orgMembership, 0, pageSize);
+                for (Study study : studies) {
+                    if (study.getIdentifier() == studyId) {
+                        sponsored = true;
+                        break;
+                    }
+                }
+                if (!sponsored) {
+                    LOG.info("User's org does not sponsor the given studyId.");
+                    return;
+                }
+            }
+
+            if (participant.getEmail() == null || !participant.isEmailVerified()) {
+                LOG.info("User does not have a validated email address.");
+                return;
+            }
+
+            int offsetBy = 0;
+            // get first page of study participants
+            List<StudyParticipant> studyParticipants = bridgeHelper.getStudyParticipantsForApp(appId, orgMembership,
+                    offsetBy, pageSize, studyId);
+
+            try (CSVWriter csvFileWriter = new CSVWriter(fileHelper.getWriter(csvFile))) {
+                writeStudyParticipants(csvFileWriter, studyParticipants, offsetBy, appId, orgMembership, studyId, 0);
+            } catch (IOException ex) {
+                throw new WorkerException("Error creating file " + csvFile + ": " + ex.getMessage(), ex);
+            }
+
+            // zip up and email the csv file
+            AppInfo appInfo = dynamoHelper.getApp(appId);
+            AccountInfo accountInfo = bridgeHelper.getAccountInfo(appId, userId);;
+
+            zipFiles(csvFile, zipFile, password);
+            sesHelper.sendEmailWithAttachmentToAccount(appInfo, accountInfo, zipFile.getAbsolutePath());
+
+        } catch (BridgeSDKException ex) {
+            int status = ex.getStatusCode();
+            if (status >= 400 && status < 500) {
+                throw new PollSqsWorkerBadRequestException(ex);
+            } else {
+                throw new RuntimeException(ex);
+            }
+        } finally {
+            cleanupFiles(csvFile, zipFile, tmpDir);
+        }
+    }
+
+    /** Write all of the study participants to the CSV file */
+    void writeStudyParticipants(CSVWriter csvFileWriter, List<StudyParticipant> studyParticipants, int offsetBy,
+                                String appId, String orgMembership, String studyId, int numParticipants) throws IOException, InterruptedException {
```

Comment:
What is numParticipants used for, and why does it need to be passed in as an input to this method?

Reply:
I confused myself when I was doing this but I was responding to the comment about adding a lot message every page. I've removed the numParticipants logic and added a simple log message.